packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -206,23 +206,20 @@ class _LocalizationsScope extends InheritedWidget {
     Key key,
     @required this.locale,
     @required this.localizationsState,
-    @required this.loadGeneration,
+    @required this.typeToResources,
     Widget child,
   }) : super(key: key, child: child) {
     assert(localizationsState != null);
+    assert(typeToResources != null);
   }
 
   final Locale locale;
   final _LocalizationsState localizationsState;
-
-  /// A monotonically increasing number that changes after localizations
-  /// delegates have finished loading new data. When this number changes, it
-  /// triggers inherited widget notifications.
-  final int loadGeneration;
+  final Map<Type, dynamic> typeToResources;
 
   @override
   bool updateShouldNotify(_LocalizationsScope old) {
-    return loadGeneration != old.loadGeneration;
+    return !identical(typeToResources, old.typeToResources);
```

Comment:
pretty sure you can just use `==`, Map doesn't overload it

Reply:
Done. Only did it because `identical` is cheaper, but I don't have numbers to tell if it matters here, so switching to `==`.

Reply:
If identical is actually cheaper than == here, then we should get the VM folks to fix that. :-)

Reply:
It's more of a language issue than it is a VM issue. `==` looks like a primitive operator, while in fact it is a virtual method. Until strong mode is a thing (and perhaps non-nullability too), `==` will remain the much slower option. One day....